pandas/core/internals/blocks.py

```diff
@@ -289,13 +303,15 @@ def getitem_block(self, slicer, new_mgr_locs=None):
         if new_mgr_locs is None:
             axis0_slicer = slicer[0] if isinstance(slicer, tuple) else slicer
             new_mgr_locs = self.mgr_locs[axis0_slicer]
+        elif not isinstance(new_mgr_locs, BlockPlacement):
+            new_mgr_locs = BlockPlacement(new_mgr_locs)
 
         new_values = self._slice(slicer)
 
         if self._validate_ndim and new_values.ndim != self.ndim:
             raise ValueError("Only same dim slicing is allowed")
 
-        return self.make_block_same_class(new_values, new_mgr_locs)
+        return type(self)._simple_new(new_values, new_mgr_locs, self.ndim)
 
```

Comment:
can you use this pattern and remove make_block_same_class entirely?

Reply:
We can definitely change most usages.  There are a few that pass `ndim` that ill need to double-check.